### DISCUSSION TITLE
NPZD with particle tracking?

### DISCUSSION THREAD

**Discussion by iuryt** [OceanBioME/OceanBioME.jl#173] [Q&A] [answered]
Hi All,

I participated on a discussion for biogeochemical modelling in the context of [S-MODE project](https://podaac.jpl.nasa.gov/S-MODE?tab=mission-objectives&sections=about%2Bdata) and started to think on how easy would be use OceanBioME for a Lagrangian NPZD model.

This could be related to a project with a PhD student and I want to make sure it is easy enough for the student to focus on the science and not on the model implementation.

What do you think?

**Comment by jagoosw** [comment]
Hi @iuryt,

I'm not sure I completely understand how a biogeochemical model would work as a Lagrangian model, but if it is something like this https://doi.org/10.1016/j.jcp.2018.01.031 where you have all of the components on every particle then I think it should be quite straight forward to implement in OceanBioME.

It would be interesting to see how this goes and I'd be happy to help if there are any implementation problems so let me know!

**Comment by johnryantaylor** [comment]
Hi @iury, This should be possible and would be very interesting. If you are going to have interactions between particles, then you would need to add code to do a search for neighboring particles. The paper that Jago linked to describes an algorithm for this.  That step is quite computationally costly, so it would be worth doing it well (but that type of algorithm is quite useful for lots of particle based problems and it would be great to have that capability in OceanBioME)On Apr 1, 2024, at 7:47 AM, Jago Strong-Wright ***@***.***> wrote:﻿
Hi @iuryt,
I'm not sure I completely understand how a biogeochemical model would work as a Lagrangian model, but if it is something like this https://doi.org/10.1016/j.jcp.2018.01.031 where you have all of the components on every particle then I think it should be quite straight forward to implement in OceanBioME.
It would be interesting to see how this goes and I'd be happy to help if there are any implementation problems so let me know!

**Comment by iuryt** [comment]
Thanks for all comments and suggestions! Since that would need implementations to the source code, I think that it might not suit best for this project. But I and other colleagues have discussed about implementing this for another project. We will post it here if we start doing some work on that. I will leave it open in case others have some ideas or updates.

**Comment by johnryantaylor** [comment]
Hi @iuryt, you could do something without particle-particle interaction by solving the NPZD equations (or some subset of them, e.g. Lagrangian P and Z and Eulerian N) along particle paths. It would still be interesting to compare that with an Eulerian approach. On Apr 1, 2024, at 11:04 AM, iury simoes-sousa ***@***.***> wrote:﻿
Thanks for all comments and suggestions! Since that would need implementations to the source code, I think that it might not suit best for this project. But I and other colleagues have discussed about implementing this for another project. We will post it here if we start doing some work on that. I will leave it open in case others have some ideas or updates.

**Reply by iuryt** [reply]
That's true. You mean to have both P and Z for each particle and use N as a tracer, right?

**Comment by johnryantaylor** [comment, marked as answer]
Yes, that’s what I was thinking.  Each particle could solve equations for both P and Z (basically treating this as a water parcel with its own ecosystem dynamics). You could do the same for nutrients too, or treat nutrients as an Eulerian scalar to account for diffusion. On Apr 1, 2024, at 12:58 PM, iury simoes-sousa ***@***.***> wrote:﻿
That's true. You mean to have both P and Z for each particle and use N as a tracer, right?
